pkg/endpoint/endpoint.go

```diff
@@ -162,6 +174,30 @@ func init() {
 	}
 }
 
+// getCiliumClient builds and returns a k8s auto-generated client for cilium
+// objects
+func getCiliumClient() (ciliumClient cilium_client_v2.CiliumV2Interface, err error) {
+	// This allows us to reuse the k8s client
+	ciliumEndpointSyncControllerOnce.Do(func() {
+		restConfig, err := k8s.CreateConfig()
+		if err != nil {
+			return
+		}
+		k8sClient, err := clientset.NewForConfig(restConfig)
+		if err != nil {
+			return
+		}
+
+		ciliumEndpointSyncControllerK8sClient = k8sClient
+	})
+
+	if err != nil {
```

Comment:
Does this really work? Won't the `:=` in the `func()` shadow `err`?

Reply:
nope, that totally doesn't work. I'll fix it. Thanks!